src/metabase/task.clj

```diff
@@ -185,12 +185,22 @@
    new-trigger :- (ms/InstanceOfClass Trigger)]
   (try
     (when-let [scheduler (scheduler)]
+      ;; TODO: a job could have multiple triggers, so the first trigger is not guaranteed to be the one we want to
+      ;; replace. Should we check that the key name is matching?
```

Comment:
This is a good question, but I'm not sure the trigger key _has_ to be the same. Any callers of this function must assume that the job has only one trigger for the function to work correctly, which is the case for most jobs we have. e.g. the send follow up emails job:
https://github.com/metabase/metabase/blob/cf928b07974c3c1d1be9b14f7e3a29e1b4f3499b/src/metabase/task/follow_up_emails.clj#L67-L77

With that in mind, requiring that the trigger's key be the same between the old the new trigger might be an unnecessary constraint on the usage of this function. Maybe it would be better to explain the assumption about the 1:1 relationship between job and trigger in the function'ss docstring, or to assert that there's only one result from `qs/get-triggers-of-job`?

Reply:
I was thinking that as well but I'm not sure having a assert and docstring is good enough to prevent footgun.

One example was when I wrote the test for trigger replacing priority, I didn't think about setting up the test so that it has 2 triggers and making sure `reschedule-trigger` will replace the correct one for me, tho. 

Yes it's true that most of our task has 1:1 relationship between job and trigger, except sync database jobs and send pulse (just now)

Reply:
Fair point. Sounds like it needs a larger refactoring to do the job properly, so yes let's leave this TODO and sort it out later.